src/index/am.rs

```diff
@@ -332,17 +368,41 @@ pub unsafe extern "C" fn amgettuple(
     if unsafe { (*(*scan).xs_snapshot).snapshot_type } != pgrx::pg_sys::SnapshotType_SNAPSHOT_MVCC {
         pgrx::error!("scanning with a non-MVCC-compliant snapshot is not supported");
     }
+    let filter = unsafe {
+        match (*scan).numberOfKeys {
+            1 => {
+                let data = (*scan).keyData.add(0);
+                let threshold = f32::from_datum((*data).sk_argument, false).unwrap();
+                match (*data).sk_strategy {
+                    2 => Some(PushdownRange {
+                        operator: RangeOperator::MeasureLess,
+                        threshold,
+                    }),
+                    3 => Some(PushdownRange {
+                        operator: RangeOperator::MeasureLessEqual,
+                        threshold,
+                    }),
+                    _ => None,
+                }
+            }
+            _ => None,
+        }
+    };
     let scanner = unsafe { (*scan).opaque.cast::<Scanner>().as_mut().unwrap_unchecked() };
     let oid = unsafe { (*(*scan).indexRelation).rd_id };
     let handle = from_oid_to_handle(oid);
-    if let Some(pointer) = am_scan::scan_next(scanner, handle) {
+    if let Some((distance, pointer)) = am_scan::scan_next(scanner, handle) {
         let ctid = pointer_to_ctid(pointer);
         unsafe {
             (*scan).xs_heaptid = ctid;
             (*scan).xs_recheck = false;
```

Comment:
If there is a key that is pushed to the index, we should
1. ignore that key and set `xs_recheck` to true
2. or, check the key in the index and set `xs_recheck` to false
So there should not be warnings.

The index must handle these cases:
1. multiple keys and no order-by key (to fix it, you must modify the cost estimate fucntion: a trick is here)
2. vectors in key and order-by key are different
3. no key and no order-by key

The index should not emit warnings or errors and **must return correct results**. `xs_recheck` is used for it.